flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -921,11 +921,20 @@ public ActorGateway createActivatorDeactivator(ActorSystem actorSystem, UUID lea
 	// ------------------------------------------------------------------------
 
 	private class ScheduledTrigger extends TimerTask {
+		private final long processingInterval;
+		private long lastCheckpoint = 0;
+
+		public ScheduledTrigger(long guaranteedInterval) {
+			this.processingInterval = guaranteedInterval;
```

Comment:
Why do we need the guaranteedInterval within the `ScheduledTrigger`? Can we be sure that the `time - lastCheckpoint` is always greater than `processingInterval` if the task is executed?